### DISCUSSION TITLE
Winget powershell install script on Windows Server

### DISCUSSION THREAD

**Discussion by nolme** [microsoft/winget-cli#4444] [Show and tell]
Hello,
based on this topic : [](https://stackoverflow.com/questions/68100663/how-do-i-install-winget-on-windows-server-2019)
I try to make a full script for using WinGet on Windows Server. All seems to be OK but winget show no output.
Any idea ?

Thanks

Here is the full code (french 1036 & english 1033 are managed)  : 

`# Paramètre pour forcer la réinstallation
$forceReinstall = $true

# Ajustement de la page de code pour l'UTF-8
chcp 1252

# Configuration de l'encodage de sortie
[Console]::OutputEncoding = [System.Text.Encoding]::UTF8

# Installation de NuGet si nécessaire ou forcé
if ($forceReinstall -or -not (Get-PackageProvider -Name NuGet) -or (Get-PackageProvider -Name NuGet).Version -lt 2.8.5.201)
{
	Write-Host "Installation de NuGet..."
	try
	{
		Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force -Confirm:$false -ForceBootstrap
	}
	catch
	{
		Write-Error -Message $_.Exception
	}
}
else
{
	Write-Host "NuGet est déjà à jour."
}

# Vérification de la présence de winget ou forcer l'installation
try
{
	if ($forceReinstall)
	{
		throw "Forcing reinstall."
	}
	winget --version
	Write-Host "Winget est déjà installé."
}
catch
{
	Write-Host "Installation de Winget..."
	# Installation de VCLibs forcée si nécessaire
	if ($forceReinstall -or -not (Get-AppxPackage -Name Microsoft.VCLibs.140.00.Desktop))
	{
		Write-Host "Installation de VCLibs..."
		Add-AppxPackage 'https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx'
	}

	# Installation de Microsoft.UI.Xaml forcée si nécessaire
	if ($forceReinstall -or -not (Get-AppxPackage -Name Microsoft.UI.Xaml.2.8))
	{
		Write-Host "Installation de Microsoft.UI.Xaml..."
		Invoke-WebRequest -Uri https://www.nuget.org/api/v2/package/Microsoft.UI.Xaml/2.8.6 -OutFile .\microsoft.ui.xaml.2.8.6.zip
		Expand-Archive .\microsoft.ui.xaml.2.8.6.zip -Force
		Add-AppxPackage .\microsoft.ui.xaml.2.8.6\tools\AppX\x64\Release\Microsoft.UI.Xaml.2.8.appx
	}

	# Installation forcée de Microsoft.DesktopAppInstaller
	Write-Host "Installation de Microsoft.DesktopAppInstaller..."
	Invoke-WebRequest -Uri https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle -OutFile .\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
	Add-AppxPackage .\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle

	# Mise à jour du chemin d'accès pour winget
	$ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
	if ($ResolveWingetPath)
	{
		$WingetPath = $ResolveWingetPath[-1].Path
		$ENV:PATH += ";$WingetPath"
		Write-Host "Chemin d'accès à Winget ajouté à PATH."
	}
	
	# Détecter la langue de l'interface utilisateur du système d'exploitation
	$osUILanguage = (Get-WmiObject -Class Win32_OperatingSystem).OSLanguage
	Write-Host "OS language : " + $osUILanguage

	# Définir les paramètres en fonction de la langue
	if ($osUILanguage -eq 1036) {
		# 1036 correspond à Français - France
		$takeownParam = "/D O"
		$icaclsGroup = "Administrateurs"
	}
	elseif ($osUILanguage -eq 1033)
	{
		# English
		$takeownParam = "/D Y"
		$icaclsGroup = "Administrators"
	}
	else
	{
		# All other cases. Use english as default
		$takeownParam = "/D Y"
		$icaclsGroup = "Administrators"
	}

	$folderMask = "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*"
	$folders = Get-ChildItem -Path $folderMask -Directory | Where-Object { $_.Name -like "*_x64_*" }
	foreach ($folder in $folders)
	{
		Write-Host "Traitement de: $folderPath"
		$folderPath = $folder.FullName
		TAKEOWN /F $folderPath /A /R $takeownParam
		ICACLS $folderPath /grant:r $icaclsGroup:F /T
		Write-Host "Permissions mises à jour pour $folderPath"
	}

	# Modifie les droits d'appartenance
	$folderMask = "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*"
	$folders = Get-ChildItem -Path $folderMask -Directory | Where-Object { $_.Name -like "*_x64_*" }
	foreach ($folder in $folders)
	{
		Write-Host "Traitement de: $folderPath"
		$folderPath = $folder.FullName
		TAKEOWN /F $folderPath /A /R $takeownParam
		ICACLS $folderPath /grant:r $icaclsGroup:F /T
		Write-Host "Permissions mises à jour pour $folderPath"
	}
}`

**Comment by MatthiasGuelck** [comment]
The 'O' for the "TAKEOWN" command and the "Administrateurs" for the "ICACLS" command only work with French as the system language

**Reply by nolme** [reply]
Hello, I update the script to manage both English & French OS. 

- The '/D Y' didn't work (invalid argument option) for TAKEOWN
- The '/grant:r' didn't work too (invalid parameter) for ICACLS

Any idea ?

**Reply by MatthiasGuelck** [reply]
1. "The '/D Y' didn't work (invalid argument option) for TAKEOWN"
Which system language did you use to test TAKEOWN with "/D Y"?
Your "else" to "if ($osUILanguage -eq 1036)" will not work.
Your script runs into this "else" branch e.g. with German as system language.
In this case you would have to use "/D J".

2. "The '/grant:r' didn't work too (invalid parameter) for ICACLS"
The same problem applies here as with 1.
In addition, you must use '"' somewhat unusually with this command if you are using a variable:
 /grant:r $icaclsGroup":F" /T 
In addition, the section with the "for" loop seems to be duplicated.
